core/src/avm2/globals/flash/utils/bytearray.rs

```diff
@@ -0,0 +1,946 @@
+use crate::avm2::activation::Activation;
+use crate::avm2::bytearray::Endian;
+use crate::avm2::class::{Class, ClassAttributes};
+use crate::avm2::method::Method;
+use crate::avm2::names::{Namespace, QName};
+use crate::avm2::object::{Object, TObject};
+use crate::avm2::string::AvmString;
+use crate::avm2::traits::Trait;
+use crate::avm2::value::Value;
+use crate::avm2::Error;
+use encoding_rs::Encoding;
+use encoding_rs::UTF_8;
+use gc_arena::{GcCell, MutationContext};
+
+/// Implements `flash.utils.ByteArray`'s instance constructor.
+pub fn instance_init<'gc>(
+    activation: &mut Activation<'_, 'gc, '_>,
+    this: Option<Object<'gc>>,
+    _args: &[Value<'gc>],
+) -> Result<Value<'gc>, Error> {
+    if let Some(this) = this {
+        activation.super_init(this, &[])?;
+    }
+
+    Ok(Value::Undefined)
+}
+
+/// Implements `flash.utils.ByteArray`'s class constructor.
+pub fn class_init<'gc>(
+    _activation: &mut Activation<'_, 'gc, '_>,
+    _this: Option<Object<'gc>>,
+    _args: &[Value<'gc>],
+) -> Result<Value<'gc>, Error> {
+    Ok(Value::Undefined)
+}
+
+/// Writes a single byte to the bytearray
+pub fn write_byte<'gc>(
+    activation: &mut Activation<'_, 'gc, '_>,
+    this: Option<Object<'gc>>,
+    args: &[Value<'gc>],
+) -> Result<Value<'gc>, Error> {
+    if let Some(mut bytearray) = this
+        .unwrap()
+        .as_bytearray_mut(activation.context.gc_context)
+    {
+        let byte = args
+            .get(0)
+            .cloned()
+            .unwrap_or(Value::Undefined)
+            .coerce_to_i32(activation)?;
+        bytearray.write_byte(byte as u8);
+    }
+
+    Ok(Value::Undefined)
+}
+
+/// Writes multiple bytes to the bytearray from another bytearray
+pub fn write_bytes<'gc>(
+    activation: &mut Activation<'_, 'gc, '_>,
+    this: Option<Object<'gc>>,
+    args: &[Value<'gc>],
+) -> Result<Value<'gc>, Error> {
+    if let Some(Value::Object(second_array)) = args.get(0) {
+        let combining_bytes = second_array.as_bytearray().unwrap().bytes().clone();
```

Comment:
We don't (yet) enforce argument types, so it's entirely possible for the second array here to be a non-`ByteArray` object. This should return an `ArgumentError` instead of a panic in that case.